src/modules/sensors/sensors.cpp

```diff
@@ -421,143 +445,173 @@ void Sensors::InitializeVehicleIMU()
+	if ((magnetometer.timestamp != 0) && (magnetometer.timestamp != _magnetometer_prev_timestamp)) {
+		_magnetometer_pub.publish(magnetometer);
+		_magnetometer_prev_timestamp = magnetometer.timestamp;
 
-		/* check analog airspeed */
-		adc_poll();
+		if (!_armed) {
+			_voted_sensors_update.calcMagInconsistency(_sensor_preflight);
+		}
+	}
 
-		diff_pres_poll(airdata);
+	if (_sensor_combined.timestamp != _sensor_combined_prev_timestamp) {
 
-		if (raw.timestamp > 0) {
+		_voted_sensors_update.setRelativeTimestamps(_sensor_combined);
+		_sensor_pub.publish(_sensor_combined);
+		_sensor_combined_prev_timestamp = _sensor_combined.timestamp;
 
-			_voted_sensors_update.setRelativeTimestamps(raw);
+		// If the the vehicle is disarmed calculate the length of the maximum difference between
+		// IMU units as a consistency metric and publish to the sensor preflight topic
+		if (!_armed) {
+			_voted_sensors_update.calcAccelInconsistency(_sensor_preflight);
+			_voted_sensors_update.calcGyroInconsistency(_sensor_preflight);
 
-			_sensor_pub.publish(raw);
+			_sensor_preflight.timestamp = hrt_absolute_time();
+			_sensor_preflight_pub.publish(_sensor_preflight);
+		}
+	}
 
-			if (airdata.timestamp != airdata_prev_timestamp) {
-				_airdata_pub.publish(airdata);
-			}
+	// keep adding sensors as long as we are not armed,
+	// when not adding sensors poll for param updates
+	if (!_armed && hrt_elapsed_time(&_last_config_update) > 500_ms) {
+		_voted_sensors_update.initializeSensors();
+		InitializeVehicleIMU();
+		_last_config_update = hrt_absolute_time();
 
-			if (magnetometer.timestamp != magnetometer_prev_timestamp) {
-				_magnetometer_pub.publish(magnetometer);
-			}
+	} else {
+		// check parameters for updates
+		parameter_update_poll();
+	}
 
-			_voted_sensors_update.checkFailover();
+	perf_end(_loop_perf);
+}
 
-			/* If the the vehicle is disarmed calculate the length of the maximum difference between
-			 * IMU units as a consistency metric and publish to the sensor preflight topic
-			*/
-			if (!_armed) {
-				preflt.timestamp = hrt_absolute_time();
-				_voted_sensors_update.calcAccelInconsistency(preflt);
-				_voted_sensors_update.calcGyroInconsistency(preflt);
-				_voted_sensors_update.calcMagInconsistency(preflt);
+int Sensors::task_spawn(int argc, char *argv[])
+{
+	bool hil_enabled = false;
+	bool error_flag = false;
 
-				_sensor_preflight.publish(preflt);
-			}
-		}
+	int myoptind = 1;
+	int ch;
+	const char *myoptarg = nullptr;
 
-		/* keep adding sensors as long as we are not armed,
-		 * when not adding sensors poll for param updates
-		 */
-		if (!_armed && hrt_elapsed_time(&last_config_update) > 500_ms) {
-			_voted_sensors_update.initializeSensors();
-			InitializeVehicleIMU();
-			last_config_update = hrt_absolute_time();
+	while ((ch = px4_getopt(argc, argv, "h", &myoptind, &myoptarg)) != EOF) {
+		switch (ch) {
+		case 'h':
+			hil_enabled = true;
+			break;
 
-		} else {
+		case '?':
+			error_flag = true;
```

Comment:
You can remove this case, right? Because `?` is not listed in the options of `px4_getopt()` and also not described in the usage.

Reply:
I'm not sure, that's the pattern we seem to have in other cases. https://github.com/PX4/Firmware/blob/4afd7a53e854d4aab3fa2461f793d8b9d99a38d6/src/drivers/gps/gps.cpp#L1188-L1190

If I was going to do anything with command line flags in this module I'd probably get rid of them entirely. We can get HIL state from vehicle_status, and the only remaining usage seems to be increasing the sensor timeout and skipping analog airspeed read (going to be extracted soon).